src/index.js

```diff
@@ -40,14 +44,35 @@ async function requestHandler (request, env, ctx) {
   const { cidPath } = ctx
   if (!cidPath) throw new Error('missing IPFS path')
 
-  console.log('get', cidPath, 'from', env.REMOTE_PEER)
+  let response
   const { searchParams } = new URL(request.url)
-
   if (searchParams.get('format') === 'raw') {
-    return await handleBlock(request, env, ctx)
+    response = await handleBlock(request, env, ctx)
   }
   if (searchParams.get('format') === 'car') {
-    return await handleCar(request, env, ctx)
+    response = await handleCar(request, env, ctx)
+  }
+  response = await handleUnixfs(request, env, ctx)
+
+  ctx.waitUntil(
+    putToCache(request, response, caches.default)
+  )
+
+  return response
+}
+
+/**
+ * Put received response to cache.
+ *
+ * @param {Request} request
+ * @param {Response} response
+ * @param {Cache} cache
+ */
+async function putToCache (request, response, cache) {
```

Comment:
Can we please do this in the middleware after `handler` has run?

Reply:
I tried, but there is no clean way of making sure it will also not cache responses for error. The only change I am seeing it to wrap `requestHandler` in a function that handles the putToCache after handler finishes. What do you think?